cmd/state-installer/cmd.go

```diff
@@ -386,7 +387,7 @@ func postInstallEvents(out output.Outputer, cfg *config.Instance, an analytics.D
 			an.EventWithLabel(AnalyticsFunnelCat, "forward-activate-default-err", err.Error())
 			return errs.Silence(errs.Wrap(err, "Could not activate %s, error returned: %s", params.activateDefault.String(), errs.JoinMessage(err)))
 		}
-	case !params.isUpdate && os.Getenv(constants.InstallerNoSubshell) != "true":
+	case !params.isUpdate && terminal.IsTerminal(int(os.Stdin.Fd())) && os.Getenv(constants.InstallerNoSubshell) != "true":
```

Comment:
Does calling int() on a uintptr do the correct cast? I'm coming from C/C++ land where casting pointers to/from ints is tricky.

Reply:
I'm not sure in all of the cases but I _believe_ that in this case it is fine. `os.Stdin.Fd()` simply casts the `int` of the file descriptor to a `uintptr`.